modules/custom-status/custom-status.php

```diff
@@ -459,747 +466,806 @@ function post_admin_header() {
+				die( esc_html( $change_error->get_error_message() ) );
+			}
 
-		// Check to make sure the status doesn't already exist
-		if ( $this->get_custom_status_by( 'slug', $status_slug ) && ( $this->get_custom_status_by( 'id', $term_id )->slug != $status_slug ) ) {
-			$change_error = new WP_Error( 'invalid', __( 'Status already exists. Please choose another name.', 'edit-flow' ) );
-			die( $change_error->get_error_message() );
-		}
+			// Check to make sure the status doesn't already exist
+			if ( $this->get_custom_status_by( 'slug', $status_slug ) && ( $this->get_custom_status_by( 'id', $term_id )->slug != $status_slug ) ) {
+				$change_error = new WP_Error( 'invalid', esc_html__( 'Status already exists. Please choose another name.', 'edit-flow' ) );
+				die( esc_html( $change_error->get_error_message() ) );
+			}
 
-		// Check to make sure the status doesn't already exist as another term because otherwise we'd get a fatal error
-		$term_exists = term_exists( sanitize_title( $status_name ), self::taxonomy_key );
-		if ( $term_exists && isset( $term_exists['term_id'] ) && $term_exists['term_id'] != $term_id ) {
-			$change_error = new WP_Error( 'invalid', __( 'Status name conflicts with existing term. Please choose another.', 'edit-flow' ) );
-			die( $change_error->get_error_message() );
-		}
+			// Check to make sure the status doesn't already exist as another term because otherwise we'd get a fatal error
+			$term_exists = term_exists( sanitize_title( $status_name ), self::taxonomy_key );
+			if ( $term_exists && isset( $term_exists['term_id'] ) && $term_exists['term_id'] != $term_id ) {
+				$change_error = new WP_Error( 'invalid', esc_html__( 'Status name conflicts with existing term. Please choose another.', 'edit-flow' ) );
+				die( esc_html( $change_error->get_error_message() ) );
+			}
 
-		// get status_name & status_description
-		$args = array(
-			'name' => $status_name,
-			'description' => $status_description,
-			'slug' => $status_slug,
-		);
-		$return = $this->update_custom_status( $term_id, $args );
-		if( !is_wp_error( $return ) ) {
-			set_current_screen( 'edit-custom-status' );
-			$wp_list_table = new EF_Custom_Status_List_Table();
-			$wp_list_table->prepare_items();
-			echo $wp_list_table->single_row( $return );
-			die();
-		} else {
-			$change_error = new WP_Error( 'invalid', sprintf( __( 'Could not update the status: <strong>%s</strong>', 'edit-flow' ), $status_name ) );
-			die( $change_error->get_error_message() );
+			// get status_name & status_description
+			$args = array(
+				'name' => $status_name,
+				'description' => $status_description,
+				'slug' => $status_slug,
+			);
+			$return = $this->update_custom_status( $term_id, $args );
+			if ( ! is_wp_error( $return ) ) {
+				set_current_screen( 'edit-custom-status' );
+				$wp_list_table = new EF_Custom_Status_List_Table();
+				$wp_list_table->prepare_items();
+				echo wp_kses_post( $wp_list_table->single_row( $return ) );
```

Comment:
Just noting - `wp_kses_post()` on a table row seems odd. Ideally `$wp_list_table->single_row()` should return sanitized HTML and we should ignore the warning here. However, I tested this (happens during quick edits on the EditFlow Admin -> Custom Statuses page, then hover over a status and click "Quick Edit" and then "Update Status"), and it seems to work totally fine.

Reply:
Yip, agreed on that. That's how I had tried it, but the resulting ignore wherever this table class is used wasn't too well liked by me. I'd like to minimize the ignores so I ended up doing this. 

That said, I'm thinking that we can see during our feature development if we are going to keep this class or do it differently perhaps.